Github container failing when commit author is anonymous

`author` can appear blank for `singleCommit` when someone commits without valid github user email setup in the github config.

Error message:
```
github            | /usr/src/app/service.js:65
github            |                         singleCommit.author.date = time.toLocaleString("en-US", {timeZone : "America/Toronto"});
github            |                                                  ^
github            | 
github            | TypeError: Cannot set property 'date' of null
github            |     at /usr/src/app/service.js:65:50
github            |     at Array.forEach (<anonymous>)
github            |     at Request._callback (/usr/src/app/service.js:61:34)
github            |     at Request.self.callback (/usr/src/app/node_modules/request/request.js:185:22)
github            |     at Request.emit (events.js:196:13)
github            |     at Request.<anonymous> (/usr/src/app/node_modules/request/request.js:1161:10)
github            |     at Request.emit (events.js:196:13)
github            |     at IncomingMessage.<anonymous> (/usr/src/app/node_modules/request/request.js:1083:12)
github            |     at Object.onceWrapper (events.js:284:20)
github            |     at IncomingMessage.emit (events.js:201:15)
github            | npm ERR! code ELIFECYCLE
github            | npm ERR! errno 1
github            | npm ERR! github-server@1.0.0 start: `node server.js`
github            | npm ERR! Exit status 1
github            | npm ERR! 
github            | npm ERR! Failed at the github-server@1.0.0 start script.
github            | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
github            | 
github            | npm ERR! A complete log of this run can be found in:
github            | npm ERR!     /root/.npm/_logs/2019-05-23T22_08_28_057Z-debug.log

```

--- a/Github/service.js
+++ b/Github/service.js
@@ -42,10 +42,8 @@ module.exports.getRepos = () => {
             );
             repo.branches_url = `${
               repo.branches_url
-            }?per_page=100&access_token=${key}`;
-            if (
-              new Date(today - new Date(repo.pushed_at)) < new Date(recency)
-            ) {
+              }?per_page=100&access_token=${key}`;
+            if (today - new Date(repo.pushed_at) < recency) {
               branchUrls.push({
                 name: repo.name,
                 url: repo.branches_url,
@@ -65,7 +63,7 @@ const getCommits = commitObject => {
       {
         url: `https://api.github.com/repos/Seneca-CDOT/${
           commitObject.repo
-        }/commits?sha=${commitObject.br.sha}&per_page=100&access_token=${key}`,
+          }/commits?sha=${commitObject.br.sha}&per_page=100&access_token=${key}`,
         headers: { 'User-Agent': 'request' },
       },
       (err, res, data) => {
@@ -74,13 +72,12 @@ const getCommits = commitObject => {
           reject(err);
         } else {
           JSON.parse(data).forEach(singleCommit => {
-            if (
-              new Date(today - new Date(singleCommit.commit.author.date)) <
-              new Date(recency)
-            ) {
-              const time = new Date(singleCommit.commit.author.date);
+            const { commit: { author } } = singleCommit;
+
+            if (today - new Date(author.date) < recency) {
+              const time = new Date(author.date);
               singleCommit.author = {
-                name: singleCommit.commit.author.name,
+                name: author.name,
                 date: time.toLocaleString('en-US', {
                   timeZone: 'America/Toronto',
                 }),

--- a/Github/service.js
+++ b/Github/service.js
@@ -1,4 +1,4 @@
-/***********************************************************
+/* **********************************************************
 // OSTEP Dashboard Github API
 // service.cpp
 // Date Created: 2018/09/22
@@ -6,129 +6,159 @@
 // Email: yzhu132@myseneca.ca
 // Description: Github API that gets all the sorted recent 
 // commits from an organization/user
-***********************************************************/
+********************************************************** */
 
+const request = require('request');
 
-var request = require("request");
-var key = process.env.GITHUB_TOKEN;
-var repoUrl = 'https://api.github.com/orgs/Seneca-CDOT/repos?per_page=100&access_token=' + key;
+const key = process.env.GITHUB_TOKEN;
+const repoUrl = `https://api.github.com/orgs/Seneca-CDOT/repos?per_page=100&access_token=${key}`;
 console.log(key);
-let branchUrls = []
+let branchUrls = [];
 let recentCommits = [];
 
-let today = new Date();
+const today = new Date();
 const recency = 24 * 60 * 60 * 1000;
 
 module.exports.getRepos = () => {
-    repoUrls = [];
-    branchUrls = [];
-    recentCommits = [];
+  branchUrls = [];
+  recentCommits = [];
 
-    return new Promise((resolve, reject) => {
-        request.get({
-            url: repoUrl,
-            headers: { 'User-Agent': 'request' },
-        }, (err, res, data) => {
-            if (err) {
-                console.log('Error:', err);
-                reject("Unable to get repos.");
-            } else {
-                let reposX = JSON.parse(data);
-                reposX.forEach(repo => {
-                    if (new Date(today - new Date(repo.pushed_at)) < new Date(recency)) {
-                        branchUrls.push({
-                            'name': repo.name,
-                            'url': repo.branches_url.slice(0, repo.branches_url.length - 9) + '?per_page=100&access_token=' + key
-                        });
-                    }
-                });
-                resolve();
+  return new Promise((resolve, reject) => {
+    request.get(
+      {
+        url: repoUrl,
+        headers: { 'User-Agent': 'request' },
+      },
+      (err, res, data) => {
+        if (err) {
+          console.log('Error:', err);
+          reject(new Error(`Unable to get repos.`));
+        } else {
+          const reposX = JSON.parse(data);
+          reposX.forEach(repo => {
+            repo.branches_url = repo.branches_url.slice(
+              0,
+              repo.branches_url.length - 9,
+            );
+            repo.branches_url = `${
+              repo.branches_url
+              }?per_page=100&access_token=${key}`;
+            if (
+              new Date(today - new Date(repo.pushed_at)) < new Date(recency)
+            ) {
+              branchUrls.push({
+                name: repo.name,
+                url: repo.branches_url,
+              });
             }
-        });
-    });
-}
-
-const getCommits = (commitObject) => {
-    return new Promise((resolve, reject) => {
-        request.get({
-            url: 'https://api.github.com/repos/Seneca-CDOT/' + commitObject.repo + '/commits?sha=' + commitObject.br.sha + '&per_page=100&access_token=' + key,
-            headers: { 'User-Agent': 'request' },
-        }, (err, res, data) => {
-            if (err) {
-                console.log('Error:', err);
-                reject(err);
-            } else {
-                JSON.parse(data).forEach(singleCommit => {
-                    if (new Date(today - new Date(singleCommit.commit.author.date)) < new Date(recency)) {
-                        // To make it work with Dashboard/src/components/Github.js
-                        const time = new Date(singleCommit.commit.author.date);
-                        singleCommit.author.date = time.toLocaleString("en-US", {timeZone : "America/Toronto"});
-                        singleCommit.author.name = singleCommit.commit.author.name;
-                        singleCommit.message = singleCommit.commit.message;
-                        singleCommit.repoName = commitObject.repo;
-                        singleCommit.branchName = commitObject.br.name;
-                        recentCommits.push(singleCommit);
-                    }
-                });
-                resolve();
-            }
-        });
-    });
+          });
+          resolve();
+        }
+      },
+    );
+  });
 };
 
-const getBranches = (branchObject) => {
-    return new Promise((resolve, reject) => {
-        request.get({
-            url: branchObject.url,
-            headers: { 'User-Agent': 'request' },
-        }, (err, res, data) => {
-            if (err) {
-                console.log('Error:', err);
-                reject(err);
-            } else {
-                let listOfBranches = {
-                    repo: branchObject.name,
-                    branches: []
-                };
-                JSON.parse(data).forEach(singleBranch => {
-                    singleBranch.repo = branchObject.name;
-                    listOfBranches.branches.push({name: singleBranch.name, sha: singleBranch.commit.sha});
-                });
-                resolve(listOfBranches);
+const getCommits = commitObject => {
+  return new Promise((resolve, reject) => {
+    request.get(
+      {
+        url: `https://api.github.com/repos/Seneca-CDOT/${
+          commitObject.repo
+        }/commits?sha=${commitObject.br.sha}&per_page=100&access_token=${key}`,
+        headers: { 'User-Agent': 'request' },
+      },
+      (err, res, data) => {
+        if (err) {
+          console.log('Error:', err);
+          reject(err);
+        } else {
+          JSON.parse(data).forEach(singleCommit => {
+            if (
+              new Date(today - new Date(singleCommit.commit.author.date)) <
+              new Date(recency)
+            ) {
+              const time = new Date(singleCommit.commit.author.date);
+              singleCommit.author = {
+                name: singleCommit.commit.author.name,
+                date: time.toLocaleString('en-US', {
+                  timeZone: 'America/Toronto',
+                }),
+              };
+              singleCommit.message = singleCommit.commit.message;
+              singleCommit.repoName = commitObject.repo;
+              singleCommit.branchName = commitObject.br.name;
+              recentCommits.push(singleCommit);
             }
-        });
-    });
+          });
+          resolve();
+        }
+      },
+    );
+  });
+};
+
+const getBranches = branchObject => {
+  return new Promise((resolve, reject) => {
+    request.get(
+      {
+        url: branchObject.url,
+        headers: { 'User-Agent': 'request' },
+      }, (err, res, data) => {
+        if (err) {
+          console.log('Error:', err);
+          reject(err);
+        } else {
+          const listOfBranches = {
+            repo: branchObject.name,
+            branches: [],
+          };
+          JSON.parse(data).forEach(singleBranch => {
+            singleBranch.repo = branchObject.name;
+            listOfBranches.branches.push({
+              name: singleBranch.name,
+              sha: singleBranch.commit.sha,
+            });
+          });
+          resolve(listOfBranches);
+        }
+      },
+    );
+  });
 };
 
 module.exports.getAllCommitsTogether = () => {
-    return new Promise((resolve, reject) => {
-        let promises = [];
-        branchUrls.forEach((branchUrl) => {
-            promises.push(getBranches(branchUrl));
+  return new Promise((resolve, reject) => {
+    let promises = [];
+    branchUrls.forEach(branchUrl => {
+      promises.push(getBranches(branchUrl));
+    });
+    Promise.all(promises)
+      .then(branchesPerRepo => {
+        promises = [];
+        branchesPerRepo.forEach(rep => {
+          rep.branches.forEach(branch => {
+            promises.push(getCommits({ repo: rep.repo, br: branch }));
+          });
         });
         Promise.all(promises)
-            .then((branchesPerRepo) => {
-                promises = [];
-                branchesPerRepo.forEach(rep =>{
-                    rep.branches.forEach(branch =>{
-                        promises.push(getCommits({repo: rep.repo, br: branch}));
-                    });
-                });
-                Promise.all(promises)
-                .then(()=>{
-                    recentCommits.sort((firstCommit, secondCommit) => {
-                        return new Date(secondCommit.commit.author.date) - new Date(firstCommit.commit.author.date);
-                    });
-                    resolve(recentCommits);
-    
-                }).catch((err)=>{
-                    console.log(err);
-                    reject(err);
-                });
-            })
-            .catch((err) => {
-                console.log(`Error: ${err}`);
-                reject(err);
+          .then(() => {
+            recentCommits.sort((firstCommit, secondCommit) => {
+              return (
+                new Date(secondCommit.commit.author.date) -
+                new Date(firstCommit.commit.author.date)
+              );
             });
-    });
+            resolve(recentCommits);
+          })
+          .catch(err => {
+            console.log(err);
+            reject(err);
+          });
+      })
+      .catch(err => {
+        console.log(`Error: ${err}`);
+        reject(err);
+      });
+  });
 };

--- a/Github/service.js
+++ b/Github/service.js
@@ -38,11 +38,11 @@ module.exports.getRepos = () => {
           reposX.forEach(repo => {
             repo.branches_url = repo.branches_url.slice(
               0,
-              repo.branches_url.length - 9,
+              repo.branches_url.indexOf('{/branch}'),
             );
             repo.branches_url = `${
               repo.branches_url
-              }?per_page=100&access_token=${key}`;
+            }?per_page=100&access_token=${key}`;
             if (
               new Date(today - new Date(repo.pushed_at)) < new Date(recency)
             ) {

